File: backend/schemas/audit.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from utils.enums import AuditModule


class AuditLogOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    module: AuditModule
    record_type: str
    record_id: int
    field_name: str
    old_value: str | None
    new_value: str | None
    user_id: int | None
    timestamp: datetime

File: frontend/src/types/audit.ts
export type AuditModule =
  | "PRODUCT"
  | "SALES_ORDER"
  | "PURCHASE_ORDER"
  | "MANUFACTURING_ORDER"
  | "BOM"
  | "INVOICE";

/** AuditLogOut from GET /audit-logs. */
export interface AuditLog {
  id: number;
  module: AuditModule;
  record_type: string;
  record_id: number;
  field_name: string;
  old_value: string | null;
  new_value: string | null;
  user_id: number | null;
  timestamp: string;
}
